packages/admin-ui/src/utils/api/api.js

```diff
@@ -1,217 +1,5 @@
-      delete node.relationships.uid;
-
-      const saveToken = await api('csrf_token');
-      options.headers.Accept = 'application/vnd.api+json';
-      options.headers['X-CSRF-Token'] = saveToken;
-      options.method = 'POST';
-      options.body = JSON.stringify({ data: node });
-      url = `/jsonapi/${node.type.replace('--', '/')}`;
-      break;
-    }
-    case 'node:save': {
-      // Set the type to the right value for jsonapi to process.
-      // @todo Ideally this should not be differnet in the first place.
-      parameters.node = {
-        ...parameters.node,
-        type: parameters.node.type.includes('--')
-          ? parameters.node.type
-          : `node--${parameters.node.type}`,
-      };
-
-      const saveToken = await api('csrf_token');
-      options.headers.Accept = 'application/vnd.api+json';
-      options.headers['X-CSRF-Token'] = saveToken;
-      options.method = 'PATCH';
-      options.body = JSON.stringify({ data: parameters.node });
-      url = parameters.node.links.self.replace(
-        process.env.REACT_APP_DRUPAL_BASE_URL,
-        '',
-      );
-      break;
-    }
-    case 'taxonomy_vocabulary': {
-      url = '/jsonapi/taxonomy_vocabulary';
-      options.headers.Accept = 'application/vnd.api+json';
-      break;
-    }
-    case 'taxonomy_term': {
-      url = `/jsonapi/taxonomy_term/${parameters.type}`;
-      options.headers.Accept = 'application/vnd.api+json';
-      break;
-    }
-    case 'user': {
-      url = `/jsonapi/user`;
-      options.headers.Accept = 'application/vnd.api+json';
-      break;
-    }
-    case 'schema': {
-      url = `/schemata/${[parameters.entityTypeId, parameters.bundle].join(
-        '/',
-      )}`;
-      break;
-    }
-    case 'schema_by_id': {
-      url = `/admin-api/entity-schema/${parameters.entityTypeId}/${
-        parameters.entityId
-      }`;
-      break;
-    }
-    case 'field_schema': {
-      url = '/jsonapi/field_config';
-      break;
-    }
-    case 'form_display': {
-      url = '/jsonapi/entity_form_display';
-      break;
-    }
-    case 'field_storage_config': {
-      url = '/jsonapi/field_storage_config';
-      break;
-    }
-    case 'admin_ui_components': {
-      url = '/jsdrupal/components';
-      break;
-    }
-    case 'admin_ui_routes': {
-      url = '/jsdrupal/routes';
-      break;
-    }
-    default:
-      break;
-  }
-
-  const data = await fetch(
-    `${process.env.REACT_APP_DRUPAL_BASE_URL}${url}${
-      queryString
-        ? `?${qs.stringify(queryString, { arrayFormat: 'brackets' })}`
-        : ''
-    }`,
-    options,
-  ).then(res => {
-    if (![200, 201, 204].includes(res.status)) {
-      throw new ApiError(res.status, res.statusText, res);
-    }
-
-    // CSRF tokens return text, not json.
-    if (options.text) {
-      return res.text();
-    }
-    return res.json();
-  });
-  return data;
-}
+const api = apiHelper.bind(null, process.env.REACT_APP_DRUPAL_BASE_URL);
```

Comment:
It works, its not nice, but hey, its progress!

Reply:
I think this small change might resolve not nice'ness?